datafusion/sqllogictest/test_files/unnest.slt

```diff
@@ -777,10 +795,13 @@ select unnest(unnest(column2)) c2, count(column3) from recursive_unnest_table gr
 [, 6] 1
 NULL 1
 
-### TODO: group by unnest struct
 query error DataFusion error: Error during planning: Projection references non\-aggregate values
 select unnest(column1) c1 from nested_unnest_table group by c1.c0;
```

Comment:
Although the error message is weird, asserting that this query fails makes sense to me.
DuckDB doesn't allow this SQL, too. I can't find other databases with this behavior. 🤔 

```sql
D CREATE TABLE unnest_table
  AS VALUES
      ([1,2,3], [7], 1, [13, 14], {'c1': 1, 'c2': 2}),
      ([4,5], [8,9,10], 2, [15, 16], {'c1':3,'c2':4}),
      ([6], [11,12], 3, null, null),
      ([12], [null, 42, null], null, null, {'c1':7,'c2':8}),
      -- null array to verify the `preserve_nulls` option
      (null, null, 4, [17, 18], null)
  ;
D select unnest(col4) u1 from unnest_table group by u1.c1;
Binder Error: Referenced table "u1" not found!
Candidate tables: "unnest_table"
LINE 1: ...st(col4) u1 from unnest_table group by u1.c1;
```